src/Microsoft.OData.Core/UriParser/Binders/DottedIdentifierBinder.cs

```diff
@@ -55,6 +55,7 @@ internal QueryNode BindDottedIdentifier(DottedIdentifierToken dottedIdentifierTo
             SingleResourceNode parentAsSingleResource = parent as SingleResourceNode;
             IEdmSchemaType childType = UriEdmHelpers.FindTypeFromModel(state.Model, dottedIdentifierToken.Identifier, this.Resolver);
             IEdmStructuredType childStructuredType = childType as IEdmStructuredType;
+            IEdmPrimitiveType childPrimitiveType = childType as IEdmPrimitiveType;

```

Comment:
Consider moving this line to just before line 86 (if (childPrimitiveType != null...), since childPrimitiveType is only used within that block.